WcaOnRails/app/views/registrations/edit.html.erb

```diff
@@ -55,7 +55,12 @@
           offering the refund payment button.
           %>
           <% if @competition.using_payment_integrations? && payment.amount_available_for_refund > 0 %>
-            <%= horizontal_simple_form_for :payment, url: registration_payment_refund_path(@registration, payment), html: { id: :form_refund } do |f| %>
+            <% if @competition.stripe_connected? %>
+              <% refund_url = registration_payment_refund_path(@registration, payment) %>
+            <% else %>
+              <% refund_url = paypal_payment_refund_path(@registration, payment) %>
```

Comment:
Re one of my other comments above: You're passing in `payment` to the route helper here, which means that the controller not only knows the registration being refinded but also the exact `RegistrationPayment`. Meaning you can just access that one `receipt` directly. Take a look at the existing Stripe redirect controller method if you need breadcrumbs.